src/sugar.js

```diff
@@ -34,47 +34,73 @@ var sugar = {
 })
 
 SVG.extend([SVG.Element, SVG.FX], {
+  // Let the user set the matrix directly
+  matrix: function (mat, b, c, d, e, f) {
+    var matrix = new SVG.Matrix(arguments.length > 1 ? [mat, b, c, d, e, f] : mat)
```

Comment:
The good thing about the Matrix constructor is: you can just go `new SVG.Matrix(mat, b, c, d, e, f, )` and it will work. The constructor will see when mat is a matrix and use it

Reply:
however we need to check if at least ONE argument was given. Because if not, we have to return the matrix. Atm this code will fail when calling it without argument.

So maybe like this?

```js
if(mat == null) return new SVG.Matrix(this) // or this.matrixify()

return this.attr('transform', new SVG.Matrix(mat, b, c, d, e, f))
```

Looks good enough I think :)

Reply:
Looks good to me
I'll change that.